services/comparison/__init__.py

```diff
@@ -280,6 +283,65 @@ async def get_impacted_files(self):
             files_in_diff,
         )
 
+    def get_reports_uploaded_count_per_flag(self) -> List[ReportUploadedCount]:
+        """This function counts how many reports (by flag) the BASE and HEAD commit have."""
+        if self._cached_reports_uploaded_per_flag:
+            # Reports may have many sessions, so it's useful to memoize this function
+            return self._cached_reports_uploaded_per_flag
+        if not self.has_head_report() or not self.has_project_coverage_base_report():
+            log.warning(
+                "Can't calculate diff in uploads. Missing some report",
+                extra=dict(
+                    has_head_report=self.has_head_report(),
+                    has_project_base_report=self.has_project_coverage_base_report(),
+                ),
+            )
+            return []
+        per_flag_dict: Dict[str, ReportUploadedCount] = dict()
+        base_report = self.comparison.project_coverage_base.report
+        head_report = self.comparison.head.report
+        ops = [(base_report, "base_count"), (head_report, "head_count")]
+        for curr_report, curr_counter in ops:
+            for session in curr_report.sessions:
+                # We ignore carryforward sessions
+                # Because not all commits would upload all flags (potentially)
+                # But they are still carried forward
+                if session.session_type == SessionType.uploaded:
+                    if session.flags == []:
+                        session.flags = [""]
```

Comment:
A session with no flag is one that is uploaded without using the (optional) `--flag` argument in the CLI or uploader. They are not incommon if you have only a simple project and don't separates your tests.

I think it's extremely unlikely that `""` is used as an actual flag because it doesn't carry any information in it. In essence there's little difference uploading with a `""` flag and no flag. I don't think we _need_  to handle it separately.
A good callout though.